docs/WALKTHROUGH.md

```diff
@@ -0,0 +1,331 @@
+# Service Catalog Demonstration Walkthrough
+
+This document outlines the basic features of the service catalog by walking
+through a basic use case.
+
+## Step 1 - Installing the Service Catalog System
+
+The service catalog is conveniently packaged as a [helm](http://helm.sh/) 
+chart for installation.
+
+The chart is located in the [charts/catalog](../charts/catalog) directory in this repository, and 
+supports a wide variety of customizations which are laid out in the README.md in that
+directory. To install the service-catalog with sensible defaults, execute this command:
+
+```console
+helm install charts/catalog --name catalog --namespace catalog
+```
+
+## Step 2 - Understand Service Catalog Components
+
+Now that the system has been deployed to our Kubernetes cluster, we can use
+`kubectl` to talk to the service catalog API server.  The service catalog API
+has four resources:
+
+- Broker Server: A server that acts as a service broker and conforms to the 
+    [Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md)
+- `Broker` Resource: a representation of the broker server in the service-catalog system
+    that indicates where a broker server is running
+- `ServiceClass`: a service offered by a particular service broker. `ServiceClass`es are created
+    in response to a `Broker` resource being submitted to the system
+- `Instance`: an instance of a `ServiceClass` provisioned by the `Broker` for that `ServiceClass`
+- `Binding`: a binding to an `Instance` which is manifested into a Kubernetes namespace
+
+These components are building blocks of the service catalog in Kubernetes from an
+API standpoint.
+
+----
+
+#### Note: accessing the service catalog
+
+Unfortunately, `kubectl` doesn't know how to speak to both the service catalog
+API server and the main Kubernetes API server without switching contexts or
+`kubeconfig` files.  For now, the best way to access the service catalog API
+server is via a dedicated `kubeconfig` file.  You can manage the kubeconfig in
+use within a directory using the `direnv` tool.
+
+----
```

Comment:
=[

I admit I hadn't heard about direnv until I read the walkthrough but I don't understand how you would add it to your workflow. Doesn't it force you to change directories whenever you want to change contexts? The majority of the walkthrough has the user inside the service-catalog directory so I must be missing something..

In any case perhaps it'd be useful to elaborate more on direnv for those of us who haven't used it before and we still need something on how to add the context in the first place.